apps/aecore/test/aec_conductor_tests.erl

```diff
@@ -381,6 +381,72 @@ test_block_publishing() ->
     ok.
 
 %%%===================================================================
+%%% Pending block tests
+%%%===================================================================
+
+block_candidate_test_() ->
+    {foreach,
+     fun() ->
+          TmpKeysDir = setup_common(),
+          {ok, _} = ?TEST_MODULE:start_link([{autostart, false}]),
+          meck:new(aec_mining, [passthrough]),
+          meck:expect(aec_mining, create_block_candidate,
+              fun(TopBlock, TopBlockState, AdjChain) ->
+                  timer:sleep(100),
+                  meck:passthrough([TopBlock, TopBlockState, AdjChain])
+              end),
+          meck:expect(aec_mining, mine,
+              fun(_, _) ->
+                  timer:sleep(3000),
+                  {error, no_solution}
+              end),
+          TmpKeysDir
+     end,
+     fun(TmpKeysDir) ->
+          meck:unload(aec_mining),
+          teardown_common(TmpKeysDir),
+          ok
+     end,
+     [
+      {"Get block candidate", fun test_get_block_candidate/0}
+     ]}.
+
+test_get_block_candidate() ->
+    assert_stopped_and_genesis_at_top(),
+    ?assertEqual({error, not_mining}, ?TEST_MODULE:get_block_candidate()),
+    {ok, MyAccount} = aec_keys:pubkey(),
+    lists:foreach(
+        fun(_) ->
+            Tx = #spend_tx{sender = MyAccount,
+                          recipient = MyAccount,
+                          nonce = 0, fee = 0},
+            {ok, STx} = aec_keys:sign(Tx),
+            ok = aec_tx_pool:push(STx, tx_received)
+        end,
+        lists:seq(1, 3)),
+    ?TEST_MODULE:start_mining(),
+    ?assertEqual({error, miner_starting}, ?TEST_MODULE:get_block_candidate()),
```

Comment:
I see this relies on aec_mining:create_block_candidate sleep 100ms. Time-based though shall always work.

Reply:
I could not think of a better way to do it. I was thinking of blocking the execution of a process until a certain message is received but aec_minig:create_block_candidate is executed in the process of one of the conductor's workers...